### PR TITLE
CMakeLists.txt: don't install desktop files without GUI

### PR DESCRIPTION
**Pull request by ConiKost**
If OpenRCT2 is compiled without GUI (for example as a dedicated server),
it makes not sense to install the desktop files.

Signed-off-by: Conrad Kostecki <conrad@kostecki.com>

### CODE REVIEW

**Comment by ConiKost** [comment]
@janisozaur Are the icons used somethere for the dedicated server? Because if not, the if block could also avoid them.

**Comment by janisozaur** [comment]
No, they're not. They are used internally by the build system to create assets inside `g2.dat`, but the installed icons are not required.

**Comment by ConiKost** [comment]
Allright, updated PR.

File: CMakeLists.txt [hunk 1]
```diff
@@ -315,18 +315,20 @@ install(FILES "${CMAKE_CURRENT_BINARY_DIR}/g2.dat" DESTINATION "${CMAKE_INSTALL_
...
-install(FILES "resources/logo/icon_x64.png" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/64x64/apps" RENAME "openrct2.png")
-install(FILES "resources/logo/icon_x96.png" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/96x96/apps" RENAME "openrct2.png")
-install(FILES "resources/logo/icon_x128.png" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/128x128/apps" RENAME "openrct2.png")
-install(FILES "resources/logo/icon_x256.png" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/256x256/apps" RENAME "openrct2.png")
-install(FILES "resources/logo/icon_flag.svg" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/scalable/apps" RENAME "openrct2.svg")
-install(FILES "distribution/linux/openrct2.desktop" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/applications")
-install(FILES "distribution/linux/openrct2-savegame.desktop" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/applications")
-install(FILES "distribution/linux/openrct2-scenario.desktop" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/applications")
-install(FILES "distribution/linux/openrct2-uri.desktop" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/applications")
+if(NOT DISABLE_GUI)
```
Comment (tupaschoal) [inline CMakeLists.txt]:
```suggestion
if (NOT DISABLE_GUI)
```
Most of the`if`s above have a whitespace between the keyword and condition

Comment (ConiKost) [reply CMakeLists.txt]: Done, thanks.

**Pull request merged**